concordium-std-derive/src/lib.rs

```diff
@@ -60,18 +57,43 @@ pub fn init(attr: TokenStream, item: TokenStream) -> TokenStream {
 
     let ast: syn::ItemFn = syn::parse(item).expect("Init can only be applied to functions.");
 
+    let mut setup_fn_optional_args = proc_macro2::TokenStream::new();
```

Comment:
Could you insert a paragraph above describing how this new section achieves what it does.

Concretely, that `setup_fn_optional_args` sets up the datastructures, and `fn_optional_args` are the arguments that will be added to the function call.

Moreover we should document what the relationship is between them.